docs/sql/snapshot_restore.txt

```diff
@@ -0,0 +1,326 @@
+
+Creating a Snapshot
+~~~~~~~~~~~~~~~~~~~
+
+Snapshots are created inside a repository and can contain any number
+of tables. The :ref:`ref-create-snapshot` statement is used to create
+a snapshots::
+
+    cr> CREATE SNAPSHOT where_my_snapshots_go.snapshot1 ALL
+    ... WITH (wait_for_completion=true, ignore_unavailable=true);
+    CREATE OK (... sec)
+
+A snapshot is referenced by the name of the repository and the
+snapshot name, separated by a dot. If used with ``ALL``, all tables of
+the cluster (except :ref:`sys-tables <system-information>`,
+:ref:`information_schema` tables and blob tables) along with all
+persistent settings and the full cluster metadata are stored inside
+the snapshot.
+
+It is possible to select any number of tables for inclusion in the
+snapshot. In this case only the necessary metadata for those tables is
+included in the snapshot and the persistent settings will be excluded.
+::
+
+    cr> CREATE SNAPSHOT where_my_snapshots_go.snapshot2 TABLE quotes, doc.locations
+    ... WITH (wait_for_completion=true);
+    CREATE OK (... sec)
+
+.. Hidden: create partitioned table
+
+    cr> CREATE TABLE parted_table (
+    ...   id int,
+    ...   value string,
+    ...   date timestamp
+    ... ) clustered into 1 shards partitioned by (date) with (number_of_replicas=0);
+    CREATE OK (... sec)
+    cr> INSERT INTO parted_table (id, value, date)
+    ... VALUES (1, 'foo', '1970-01-01'), (2, 'bar', '2015-10-19');
+    INSERT OK, 2 rows affected (... sec)
+    cr> REFRESH TABLE parted_table;
+    REFRESH OK (... sec)
+
+Even single partitions of :ref:`partitioned_tables` can be selected
+for backup. This is especially useful if old partitions must be
+deleted but it should be possible to restore them if needed::
+
+    cr> CREATE SNAPSHOT where_my_snapshots_go.snapshot3 TABLE
+    ...    locations,
+    ...    parted_table PARTITION (date='1970-01-01')
+    ... WITH (wait_for_completion=true);
+    CREATE OK (... sec)
+
+Snapshots are **incremental**. Snapshots of the same cluster that were
+created later only store the data that is not already contained in the
+repository.
+
+All examples above are used with the argument ``wait_for_completion``
+set to *true*. As also described at the :ref:`ref-create-repository`
+reference documentation, by doing this, the statement will only respond
+(either sucessfully or not) when the snapshot is fully
+created. Otherwise the snapshot will be created in the background and
+the statement will immediately respond successfully. The status of a
+created snapshot can be retrieved by querying the :ref:`sys.snapshots
+<sys-snapshots>` system table.
+
+Restore
+-------
+
+Once a snapshot is created, it can be used to restore its tables to
+the state when the snapshot was created.
+
+To get basic information about snapshots the :ref:`sys.snapshots
+<sys-snapshots>` table can be queried::
+
+    cr> SELECT "repository", name, state, concrete_indices
```

Comment:
do we need to reserve this name? if not add to non-reserved and remove quotes. also for snapshot